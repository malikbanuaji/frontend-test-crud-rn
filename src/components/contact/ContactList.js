import React from 'react';
import {FlatList, View} from 'react-native';
import {Spacing} from '../../styles';
import ContactListItem from '../common/ContactListItem';

function ContactList({refreshing, onRefresh, data, onPressItem}) {
  function _renderItem({item}) {
    return (
      <ContactListItem
        id={item.id}
        firstName={item.firstName}
        lastName={item.lastName}
        age={item.age}
        photo={item.photo}
        onPress={onPressItem}
      />
    );
  }

  function _keyExtractor(item) {
    return item.id;
  }

  return (
    <FlatList
      testID={'contact-list'}
      refreshing={refreshing}
      onRefresh={onRefresh}
      data={data}
      renderItem={_renderItem}
      keyExtractor={_keyExtractor}
      ListFooterComponent={<View style={{marginBottom: Spacing.largest}} />}
    />
  );
}

export default ContactList;
