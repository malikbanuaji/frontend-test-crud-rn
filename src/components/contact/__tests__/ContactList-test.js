import React from 'react';

import renderer from 'react-test-renderer';
import ContactList from '../ContactList';

test('renders correctly when empty', () => {
  const onRefreshMock = jest.fn();
  const onPressItemMock = jest.fn();

  const tree = renderer
    .create(
      <ContactList
        refreshing={false}
        onRefresh={onRefreshMock}
        onPressItem={onPressItemMock}
        data={[]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
