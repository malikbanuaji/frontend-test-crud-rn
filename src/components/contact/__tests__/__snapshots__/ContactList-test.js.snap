// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when empty 1`] = `
<RCTScrollView
  ListFooterComponent={
    <View
      style={
        Object {
          "marginBottom": 28,
        }
      }
    />
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onRefresh={[MockFunction]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[MockFunction]}
      refreshing={false}
    />
  }
  refreshing={false}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  testID="contact-list"
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "marginBottom": 28,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
